connect/runtime/src/main/java/org/apache/kafka/connect/util/TopicAdmin.java

```diff
@@ -66,25 +71,55 @@
         /**
          * Specify the desired number of partitions for the topic.
          *
-         * @param numPartitions the desired number of partitions; must be positive
+         * @param numPartitions the desired number of partitions; must be positive, or -1 to
+         *                      signify using the broker's default
          * @return this builder to allow methods to be chained; never null
          */
         public NewTopicBuilder partitions(int numPartitions) {
+            if (numPartitions == NO_PARTITIONS) {
```

Comment:
this check seems to replicate a bit what's happening in `NewTopic`. Should we just pass the value, given that we have validations too already?